src/test/java/reporter/CustomReporter.java

```diff
@@ -0,0 +1,117 @@
+package reporter;
+
+import TestRailIntegration.TestRailIntegration;
+import com.gurock.testrail.APIClient;
+import lombok.SneakyThrows;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.testng.*;
+import org.testng.xml.XmlSuite;
+import slackIntegration.EyesSlack;
+
+import java.util.*;
+
+public class CustomReporter extends TestRailIntegration implements IReporter {
+
+    private static Map<Integer, String> idAndComment = new HashMap<>();
+
+    static {
+        idAndComment.put(1, "Current status is : Passed");
+        idAndComment.put(5, "Current status is : Failed");
+    }
+
+    private final Map<Integer, Long> statusIdAndCaseId = new HashMap<>();
+    private final Map<Integer, List<String>> passedTC = new HashMap<>();
+    private final Map<Integer, List<String>> failedTC = new HashMap<>();
+    private final Map<Integer, List<String>> skippedTC = new HashMap<>();
+
+
+    @SneakyThrows
+    @Override
+    public void generateReport(List<XmlSuite> xmlSuites, List<ISuite> suites,
```

Comment:
It's a Monster method that is responsible for: 
- parsing Suites Result,
- sending Status to Slack,
- API Client config,
- creating JSON body,
....
please split it and create a separate Reporter for TestRails & Slack

Reply:
Done